versioned_docs/version-2021fa/lecture10.md

```diff
@@ -0,0 +1,159 @@
+---
+id: lecture10
+title: Lecture 10
+---
+
+[Lecture Slides](https://docs.google.com/presentation/d/1tVozUya-oCwHNIHjMt926mJxvxdInIAWSN_4j19a94I/edit?usp=sharing)
+
+[Final Project Instructions](/docs/finalproject)
+
+[Final Project - Milestone 2](/docs/finalproject#milestone-2) due **12/2 by 11:59pm**
+
+[Final Project - Milestone 3](/docs/finalproject#milestone-3) due **12/9 by 11:59pm**
+
+## React Native
+
+### What is React Native?
+
+React Native allows for cross platform mobile development using a webdev framework we already know--React!
+
+React Native allows you to build UIs independent of the platform. Usually when developing an app you have to develop an Android version (using Java/Kotlin) and iOS version (using Objective-C/Swift) separately. React Native takes care of this conversion for you.
+
+### Core React Native Components
+
+Since React Native is really just React, many of the same concepts (`useState`, `props`, React Hooks, etc) still apply to React. However, instead of HTML we have `Views`. A view is the basic building block of UI in mobile development. Views can display images, hold text, handle user input, etc.
+
+Some core React Native components are:
+
+- `<View>`: A container that supports layout with flexbox, style, some touch handling, and accessibility controls. Similar to a non-scrolling `<div>`.
+- `<Text>`: Displays, styles, and nests strings of text and even handles touch events. Similar to a `<p>`
+- `<Image>`: Displays images like `<img>`
+- `<ScrollView>`: A generic scrolling container than can hold nested components and views. Similar to a `<div>`.
+- `<TextInput>`: User text input field similar to `<input type="text" />`.
+- ... and you can also define your own custom components (and use those built by the community)!
+
+### How to start a React Native Project?
+
+A popular way to use React Native is through the `Expo` framework, which allows developing, building, and iterating on iOS, Android and webapps. Expo provides a UI for you to view your changes and if you download the Expo app ([Android](https://play.google.com/store/apps/details?id=host.exp.exponent&hl=en_US), [iOS](https://apps.apple.com/us/app/expo-client/id982107779)) you can see those changes on your phone as well! After all, we're doing mobile development. Expo also has a lot of packages specific to accessing features on mobile devices like the status bar, battery level, I/O devices, etc.
+
+To start a React Native project run the following:
+
+```bash
+yarn global add expo-cli
+expo init <project name>
+yarn start
+```
+
+`expo init` is similar to `create-react-app` in that it generates boilerplate code for you.
+
+### Demo
+
+As part of the demo we built the simple TODO list app from assignment 4 and battery tracker in React Native! The code is here:
+
+```tsx title="App.tsx"
+import { addBatteryLevelListener, getBatteryLevelAsync } from 'expo-battery';
```

Comment:
Let's mention that you should do `expo install expo-battery` to install expo packages (as opposed to `yarn add` or `npm install`).
It prevents you from installing incompatible versions w/ your particular Expo SDK version